File: backend/app/schemas/portfolio.py
from pydantic import BaseModel


class PortfolioAllocation(BaseModel):
    category: str
    percentage: float
    amount: float
    rationale: str


class PortfolioScenarioResult(BaseModel):
    portfolio_type: str
    total_budget: float
    allocations: list[PortfolioAllocation]
    summary: str

File: frontend/src/types/portfolio.ts
export type PortfolioAllocation = {
  category: string;
  percentage: number;
  amount: number;
  rationale: string;
};

export type PortfolioScenarioResponse = {
  portfolio_type: string;
  total_budget: number;
  allocations: PortfolioAllocation[];
  summary: string;
};